js/app.js

```diff
@@ -1,5 +1,45 @@
-angular.module('GiphyApp', [])
+var app = angular.module('GiphyApp', ['infinite-scroll']);
 
-.run(function() {
-  console.log('App has loaded!');
-});
+app.controller("SearchCtrl", ['$scope', '$http', function($scope, $http) {
+  $scope.searchTerm='';
+
+  $scope.gifs=[];
+
+  $scope.search = function() {
+    var req = {
+      url: 'http://api.giphy.com/v1/gifs/search',
+      method: 'GET',
+      params: {
+        q: $scope.searchTerm,
+        api_key: "dc6zaTOxFJmzC"
+      }
+    };
+
+    $http(req).then(function success(res) {
+      $scope.gifs = res.data.data;
+      console.log($scope.gifs);
+    }, function error(res) {
+      console.log(res);
+    });
+  }
+
+  $scope.scrollMore = function() {
+    var req = {
+      url: 'http://api.giphy.com/v1/gifs/search',
+      method: 'GET',
+      params: {
+        q: $scope.searchTerm,
+        api_key: "dc6zaTOxFJmzC",
+        offset: 25
+      }
+    };
+
+    $http(req).then(function success(res) {
+      $scope.gifs = $scope.gifs.push(res.data.data);
```

Comment:
Hi Evan, just as you said in your second comment, you can't push an array onto an array. You'll need to iterate over it and push each item like this:
```
      res.data.data.forEach(function(item){
        $scope.gifs.push(item)
      });
```
Now it works. Also, using ```.concat``` as you mentioned is another way.